conversions/octal_to_hexadecimal.py

```diff
@@ -0,0 +1,73 @@
+def octal_to_hex(octal: str) -> str:
+    """
+    Convert an Octal number to Hexadecimal number.
+
+    >>> octal_to_hex("100")
+    '0X40'
+    >>> octal_to_hex("235")
+    '0X9D'
+    >>> octal_to_hex(17)
+    Traceback (most recent call last):
+        ...
+    TypeError: Expected a string as input
+    >>> octal_to_hex("Av")
+    Traceback (most recent call last):
+        ...
+    ValueError: Not a Valid Octal Number
+    >>> octal_to_hex("")
+    Traceback (most recent call last):
+        ...
+    ValueError: Empty string was passed to the function
+
+    For more information: https://en.wikipedia.org/wiki/Octal
+    """
+
+    if not isinstance(octal, str):
+        raise TypeError("Expected a string as input")
+    if octal.startswith("0o"):
+        octal = octal[2:]
+    if octal == "":
+        raise ValueError("Empty string was passed to the function")
+    for char in octal:
+        if char not in "01234567":
+            raise ValueError("Not a Valid Octal Number")
+
+    decimal = 0
+    for char in str(octal):
```

Comment:
```suggestion
    for char in octal:
```
`octal` is already a string